x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -0,0 +1,470 @@
+        XContentParser parser = context.parser();
+        if (parser.currentToken() == XContentParser.Token.VALUE_NULL) {
+            return;
+        }
+        XContentLocation xContentLocation = parser.getTokenLocation();
+        final SemanticTextField field;
+        boolean isWithinLeaf = context.path().isWithinLeafObject();
+        try {
+            context.path().setWithinLeafObject(true);
+            field = SemanticTextField.parse(parser, new Tuple<>(name(), context.parser().contentType()));
+        } finally {
+            context.path().setWithinLeafObject(isWithinLeaf);
+        }
+        final String fullFieldName = fieldType().name();
+        if (field.inference().inferenceId().equals(fieldType().getInferenceId()) == false) {
+            throw new DocumentParsingException(
+                xContentLocation,
+                Strings.format(
+                    "The configured %s [%s] for field [%s] doesn't match the %s [%s] reported in the document.",
+                    INFERENCE_ID_FIELD.getPreferredName(),
+                    field.inference().inferenceId(),
+                    fullFieldName,
+                    INFERENCE_ID_FIELD.getPreferredName(),
+                    fieldType().getInferenceId()
+                )
+            );
+        }
+        final SemanticTextFieldMapper mapper;
+        if (fieldType().getModelSettings() == null) {
+            context.path().remove();
+            Builder builder = (Builder) new Builder(simpleName(), fieldType().indexVersionCreated).init(this);
+            try {
+                mapper = builder.setModelSettings(field.inference().modelSettings())
+                    .setInferenceId(field.inference().inferenceId())
+                    .build(context.createDynamicMapperBuilderContext());
+                context.addDynamicMapper(mapper);
+            } finally {
+                context.path().add(simpleName());
+            }
+        } else {
+            Conflicts conflicts = new Conflicts(fullFieldName);
+            canMergeModelSettings(field.inference().modelSettings(), fieldType().getModelSettings(), conflicts);
+            try {
+                conflicts.check();
+            } catch (Exception exc) {
+                throw new DocumentParsingException(
+                    xContentLocation,
+                    "Incompatible model settings for field ["
+                        + name()
+                        + "]. Check that the "
+                        + INFERENCE_ID_FIELD.getPreferredName()
+                        + " is not using different model settings",
+                    exc
+                );
+            }
+            mapper = this;
+        }
+        var chunksField = mapper.fieldType().getChunksField();
+        var embeddingsField = mapper.fieldType().getEmbeddingsField();
+        for (var chunk : field.inference().chunks()) {
+            XContentParser subParser = XContentHelper.createParserNotCompressed(
```

Comment:
It's good to close parsers after you're done with them, saves some memory from pooled resources managed by Jackson :)

Reply:
Sure - let me add that, thanks! (f8dfd9c24ea789c52bc2c6110e04e40f51cae928)